### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by vitillo**
See the individual commits for the changes applied. This is the first of a series of PRs aimed at cleaning up the codebase.

### CODE REVIEW

**Comment by vitillo** [comment]
@mreid-moz ?

File: build.sbt [hunk 1]
```diff
@@ -14,19 +8,22 @@ lazy val root = (project in file(".")).
     retrieveManaged := true,
     ivyScala := ivyScala.value map { _.copy(overrideScalaVersion = true) },
     libraryDependencies += "org.apache.avro" % "avro" % "1.7.7",
-    libraryDependencies += "org.apache.parquet" % "parquet-avro" % "1.8.1",
+    libraryDependencies += "org.apache.parquet" % "parquet-avro" % "1.7.0",
```
Comment (mreid-moz) [inline build.sbt]: Should we add a comment about the weirdness with 1.8.1 vs. 1.7.0 from last time?

Comment (vitillo) [inline build.sbt]: I don't think so, using 1.8.1 was a mistake as we should use the same versions used by Spark.

File: src/main/scala/avro/JSON2Avro.scala [hunk 2]
```diff
@@ -24,12 +23,12 @@ object JSON2Avro{
     Some(record)
   }
 
-  def parseArray(schema : Schema, json: JValue): Option[Array[Any]] = json match {
+  def parseArray(schema : Schema, json: JValue): Option[java.util.Collection[Any]] = json match {
     case JArray(json) =>
```
Comment (mreid-moz) [inline src/main/scala/avro/JSON2Avro.scala]: Rename this `json` var to something else so we don't shadow the function parameter (maybe `elems`?)

File: src/main/scala/streams/Longitudinal.scala [hunk 3]
```diff
@@ -456,10 +456,11 @@ case class Longitudinal() extends DerivedStream {
     builder.endRecord()
   }
 
+
```
Comment (mreid-moz) [inline src/main/scala/streams/Longitudinal.scala]: Remove this line for style consistency.

File: src/main/scala/avro/JSON2Avro.scala [hunk 4]
```diff
@@ -24,12 +23,12 @@ object JSON2Avro{
     Some(record)
   }
 
-  def parseArray(schema : Schema, json: JValue): Option[Array[Any]] = json match {
+  def parseArray(schema : Schema, json: JValue): Option[java.util.Collection[Any]] = json match {
```
Comment (mreid-moz) [inline src/main/scala/avro/JSON2Avro.scala]: nit: Add a comment that we're using `java.util.Collection` for compatibility?

File: src/main/scala/streams/Longitudinal.scala [hunk 5]
```diff
@@ -829,12 +746,9 @@ case class Longitudinal() extends DerivedStream {
       JSON2Avro("payload.info",               List("subsessionId"),             "subsession_id", sorted, root, schema)
       JSON2Avro("payload.info",               List("subsessionLength"),         "subsession_length", sorted, root, schema)
       JSON2Avro("payload.info",               List("timezoneOffset"),           "timezone_offset", sorted, root, schema)
-      keyedHistograms2Avro(sorted, root, schema)
       histograms2Avro(sorted, root, schema)
-      threadHangStats2Avro(sorted, root, schema)
```
Comment (mreid-moz) [inline src/main/scala/streams/Longitudinal.scala]: Why remove `threadHangStats`?

File: src/test/scala/Longitudinal.scala [hunk 6]
```diff
@@ -1,126 +1,110 @@
...
-           ("values" -> ("1" -> 42)) ~
-           ("sum"    -> 42)) ~
-        ("GC_BUDGET_MS" ->
-           ("values" -> ("1" -> 42)) ~
-           ("sum"    -> 42)) ~
-        ("GC_MS" ->
-           ("values" -> ("1" -> 42)) ~
-           ("sum"    -> 42))
+          ("values" -> ("0" -> 0)) ~
+            ("sum" -> 0)) ~
```
Comment (mreid-moz) [inline src/test/scala/Longitudinal.scala]: The nesting was correct before - `values` and `sum` should be at the same depth.

File: src/test/scala/Longitudinal.scala [hunk 7]
```diff
@@ -1,126 +1,110 @@
...
-        ("applicationId"   -> "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}") ~
-        ("applicationName" -> "Firefox") ~
-        ("architecture"    -> "x86-64") ~
-        ("buildId"         -> "20160101001100") ~
-        ("version"         -> "46.0a2") ~
-        ("vendor"          -> "Mozilla") ~
-        ("platformVersion" -> "46.0a2") ~
-        ("xpcomAbi"        -> "x86_64-gcc3")
+        ("applicationId" -> "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}") ~
+          ("applicationName" -> "Firefox") ~
```
Comment (mreid-moz) [inline src/test/scala/Longitudinal.scala]: Correct indentation

File: src/test/scala/Longitudinal.scala [hunk 8]
```diff
@@ -130,275 +114,238 @@ class LongitudinalTest extends FlatSpec with Matchers with PrivateMethodTester{
...
+      private val path = ParquetFile.serialize(List(record).toIterator, schema)
+      private val filename = path.toString().replace("file:", "")
+
+      private val sparkConf = new SparkConf().setAppName("Longitudinal")
+      sparkConf.setMaster(sparkConf.get("spark.master", "local[1]"))
+      private val sc = new SparkContext(sparkConf)
+      sc.setLogLevel("WARN")
+      private val sqlContext = new SQLContext(sc)
+      val rows = sqlContext.read.load(filename).collect()
+      val row =  rows(0)
```
Comment (mreid-moz) [inline src/test/scala/Longitudinal.scala]: Should probably check that we have the correct number of rows, and that at least one of the fields in the resulting data is correct.

**Comment by mreid-moz** [comment]
Generally the indentation in `src/test/scala/Longitudinal.scala` should be cleaned up.

File: src/main/scala/streams/Longitudinal.scala (hunk 5 above)
Comment (vitillo) [inline src/main/scala/streams/Longitudinal.scala]: It's not working currently plus nobody is using it.

File: src/test/scala/Longitudinal.scala (hunk 6 above)
Comment (vitillo) [inline src/test/scala/Longitudinal.scala]: Argh, IntelliJ changed the formatting.

File: src/test/scala/Longitudinal.scala (hunk 8 above)
Comment (vitillo) [inline src/test/scala/Longitudinal.scala]: It's being checked later on, this is just the fixture.

**Comment by mreid-moz** [comment]
Looks good to me now. Has this been tested on live data?

**Comment by vitillo** [comment]
Yes it has.

**Pull request merged**